### PR TITLE
Use last model layer for benchmarking (do not pick softmax layer, which may be located somewhere in the middle of the model (#147490)

### PR DESCRIPTION
**Pull request by alpkn**
Use last model layer for benchmarking (do not pick softmax layer, which may be located somewhere in the middle of the model (#147490)

Ticket: [147490]

Description:
Classification Benchmark C++ classification_benchmark_demo provides incorrect classification results for some networks.

Networks affected:
levit-128s
resnest-50-pytorch
swin-tiny-patch4-window7-224
t2t-vit-14

Affected demos:
classification_benchmark_demo

### CODE REVIEW

File: demos/common/cpp/models/src/classification_model.cpp [hunk 1]
```diff
@@ -158,19 +158,16 @@ void ClassificationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model
...
-    std::shared_ptr<ov::Node> softmaxNode;
-    if (softmaxNodeIt == nodes.end()) {
-        auto logitsNode = model->get_output_op(0)->input(0).get_source_output().get_node();
-        softmaxNode = std::make_shared<ov::op::v1::Softmax>(logitsNode->output(0), 1);
-    } else {
-        softmaxNode = *softmaxNodeIt;
+
+    if (model->get_output_size() != 1) {
+        throw std::logic_error("Expected model's number of output equal to 1 (but found: " + std::to_string(model->get_output_size()) + ")");
     }
```
Comment (Wovchena) [inline demos/common/cpp/models/src/classification_model.cpp:164]:
```suggestion
```
Already verified above : `if (model->outputs().size() != 1) {`

File: demos/common/cpp/models/src/classification_model.cpp [hunk 2]
```diff
@@ -158,19 +158,16 @@ void ClassificationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model
...
+
+    auto logitsNode = model->get_output_op(0)->input(0).get_source_output().get_node();
```
Comment (Wovchena) [inline demos/common/cpp/models/src/classification_model.cpp:166]: There are models which already have SoftMax, which names I don't remember. So this already may be SoftMax.

File: demos/common/cpp/models/src/classification_model.cpp [hunk 3]
```diff
@@ -158,19 +158,16 @@ void ClassificationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model
...
+
     const auto k = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, std::vector<size_t>{nTop});
+    std::shared_ptr<ov::Node> softmaxNode = std::make_shared<ov::op::v1::Softmax>(logitsNode->output(0), 1);
```
Comment (Wovchena) [inline demos/common/cpp/models/src/classification_model.cpp:169]: Why would you move SoftMax construction to the middle of area that constructs topk. Move it above.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: demos/common/cpp/models/src/classification_model.cpp (hunk 1 above)
Comment (alpkn) [reply demos/common/cpp/models/src/classification_model.cpp]: noticed, will update that

File: demos/common/cpp/models/src/classification_model.cpp (hunk 2 above)
Comment (alpkn) [reply demos/common/cpp/models/src/classification_model.cpp:161]: Throughout 62 models from the supported [list](https://github.com/openvinotoolkit/open_model_zoo/blob/master/demos/classification_benchmark_demo/cpp/models.lst) there are 62 models with last layer of type "Result". They all work well with the change.

File: demos/common/cpp/models/src/classification_model.cpp (hunk 3 above)
Comment (alpkn) [reply demos/common/cpp/models/src/classification_model.cpp:163]: Guess it's about the order of lines

**Review by alpkn — COMMENTED** [review]
Updated

**Review by Wovchena — APPROVED** [review]
